python/cudf/cudf/tests/test_concat.py

```diff
@@ -587,3 +587,93 @@ def test_concat_dataframe_with_multiIndex(df1, df2):
     actual = pd.concat([pdf1, pdf2], axis=1)
 
     assert_eq(expected, actual)
+
+
+@pytest.mark.parametrize("ignore_index", [True, False])
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("join", ["inner", "outer"])
+@pytest.mark.parametrize("axis", [0, 1])
+def test_concat_join(ignore_index, sort, join, axis):
+    pdf1 = pd.DataFrame(
+        {
+            "x": range(10),
+            "y": list(map(float, range(10))),
+            "z": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        }
+    )
+    pdf2 = pd.DataFrame(
+        {"x": range(10, 20), "y": list(map(float, range(10, 20)))}
+    )
+    pdf3 = pd.DataFrame({"v": [1, 2], "x": [1, 2], "y": [1, 2], "z": [1, 2]})
+    pdf4 = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    pdf5 = pd.DataFrame({"c": [7, 8, 9], "d": [10, 11, 12]})
+
+    gdf1 = gd.from_pandas(pdf1)
+    gdf2 = gd.from_pandas(pdf2)
+    gdf3 = gd.from_pandas(pdf3)
+    gdf4 = gd.from_pandas(pdf4)
+    gdf5 = gd.from_pandas(pdf5)
+    # Make empty frame
+    gdf_empty1 = gdf2[:0]
+    assert len(gdf_empty1) == 0
+    pdf_empty1 = gdf_empty1.to_pandas()
+
+    assert_eq(
+        pd.concat(
+            [pdf1, pdf2], sort=sort, join=join, ignore_index=ignore_index
+        ),
+        gd.concat(
+            [gdf1, gdf2], sort=sort, join=join, ignore_index=ignore_index
+        ),
+    )
+
+    assert_eq(
+        pd.concat(
+            [pdf1, pdf2, pdf3, pdf_empty1],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+        ),
+        gd.concat(
+            [gdf1, gdf2, gdf3, gdf_empty1],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+        ),
+    )
```

Comment:
Can we split this off into a separate test case? If something starts failing due to empty objects it would make it easier to triage.

Reply:
Sounds good I'll split up the test cases you mentioned :D

Reply:
I'm still having trouble with the ignore_index case. I'll keep working on this today